python/ql/lib/semmle/python/frameworks/data/internal/ApiGraphModelsSpecific.qll

```diff
@@ -53,7 +53,7 @@ API::Node getExtraSuccessorFromNode(API::Node node, AccessPathToken token) {
   result = node.getMember(token.getAnArgument())
   or
   token.getName() = "Instance" and
-  result = node.getReturn() // commonly used Token. In Python `Instance` is just an alias for `ReturnValue`
+  result = node.getASubclass*().getReturn() // In Python `Instance` is just an alias for `Subclass.ReturnValue`
```

Comment:
> Do we have any precedence from other languages on this?

Yes, Ruby: https://github.com/github/codeql/blob/7abb7552a757d575d5b0d31de39e75451e49b582/ruby/ql/lib/codeql/ruby/ApiGraphs.qll#L103  

(The `getASubclass` predicate in Ruby is transitive and reflective).  

Python API-graphs don't have an "Instance" edge, so this subclass-lookup has to happen in MaD instead of in API-graphs like Ruby.  

> For me the main argument is that with this automatic-subclass semantics, you can't write code that targets instances of a specific class, without also getting the subclasses.  

Actually, you can do that in Python.  
The `ReturnValue` and `Instance` tokens are aliases of each other, except for the subclass thing (because Python doesn't have `new` calls).   
So you can use the `ReturnValue` token if you want instances of a specific class without also getting the subclasses.

Reply:
After struggling with how to deal with this, I think I've been turned, and want to support implicit `getASubclass` steps, not just for MaD, but also for general API graph usage. We discussed it internally in Python team, and we want to add these not just for `getReturn`/`getInstance`, but also for `getMember` and probably also `getACall`.

With that in mind, I would like to suggest we remove the `getASubclass` part for this PR, and add the implicit subclass steps all together in an other PR, so we keep consistency.

```suggestion
  result = node.getReturn() // In Python `Instance` is just an alias for `ReturnValue`
```

(I guess this suggestion might mess up changes, but not 100% sure).

Reply:
> Should I try to do the "add subclasses everywhere in API-graphs"-PR, or do you guys want that?

I'd like to do a few experiments in regards to `.getACall`, so let me try to do it :+1: